src/taipy/core/config/version_migration_config.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from copy import copy
+from typing import Any, Callable, Dict, Optional
+
+from taipy.config._config import _Config
+from taipy.config.common._template_handler import _TemplateHandler as _tpl
+from taipy.config.config import Config
+from taipy.config.section import Section
+
+from .._version._version_manager import _VersionManager
+
+
+class VersionMigrationConfig(Section):
```

Comment:
Could it be a UniqueSection holding a dictionary of migration functions? We may gain in terms of readability and maintainability.
What do you think?

Reply:
There maybe multiple Migration Config when there is multiple versions on production (0.9 -> 1.0 -> 1.1 -> ...).

I'm not sure if I understand your point correctly.

Reply:
What I had in mind is to have a single UniqueSection called MigrationConfig. It makes more sense to me since the functions are not independent; they are linked to each other by the version number. Plus, as a client, it is simpler to have a single entry point to retrieve all the functions.

This MigrationConfig could hold all the migration functions for all the versions.
it could be stored in a variable as a `Dict[targetVersionNumber, dict[config_id, function]]`
for example.
We could implement some smart accessors, like 
```
def get_functions(versionNumber: str, config_id: str) -> List[Callable]:
```

`Config.migrationConfig.get_functions("0.9", "input_data")` can return all the functions to migrate data nodes from config input_data from version 0.9 to the latest version. Meaning all functions to use to migrate data nodes to 1.0 version then to 1.1 version. So when taipy reads an entity, we just need to execute the functions returned by `Config.migrationConfig.get_functions(entity.version_number, entity.config_id)` in the correct order.

Does it make sense?